contracts/mocks/ArraysImpl.sol

```diff
@@ -16,4 +16,11 @@ contract ArraysImpl {
     function findUpperBound(uint256 element) external view returns (uint256) {
         return _array.findUpperBound(element);
     }
+
+    function sort(uint256[] memory array) external view returns (uint256[] memory sorted, uint256 gasUsed) {
+        gasUsed = gasleft();
```

Comment:
Please don't use that gas estimation trick in the repo. That is ok for benchmarking in remix, but not here.

Reply:
I would recommend doing a transaction (we can force doing a transaction to a view function), and check the gas usage. Of the entire function (the offset should always be the same). Using the `REPORT=true` will also show up some values.

Reply:
If there is better tooling that works with hardhat, we would love to try it out.

Reply:
I've enforced sending transactions in tests and here are the result of `enableGasReport=true REPORT=true npm test`:
```
|  Contract    ·  Method  ·  Min        ·  Max        ·  Avg        ·  # calls      ·  usd (avg)  │
···············|··········|·············|·············|·············|···············|··············
|  ArraysImpl  ·  sort    ·      22226  ·     216532  ·      75205  ·           13  ·          -  │
···············|··········|·············|·············|·············|···············|··············
```
It's not very useful, it still creates vague statistics from all the test runs. It also adds all kinds of overheads from calldata handling. For the largest case it's 52K gas or 31% and it's probably rather consistent, but for the cheapest case it's 22K gas or a 322-fold increase, which makes any benchmarks obscured by an elephant. If we move the test data from calldata to storage, it'll be even more dramatic.

What's wrong with in-EVM gas measurements for this particular function? I understand that things like storage warmness or memory allocations have hard to predict costs, but here we aren't doing any such things.

Reply:
This enableGasReport is indeed very rought, and its mostly used to check the delta when modifying code.
For new code, and comparing designs,  @frangio is working on another worklow.

Reply:
Removed, if anybody needs them back, they can revert the removal commit of just copy the code like frangio did.